.github/workflows/nightly-release.yaml

```diff
@@ -0,0 +1,76 @@
+name: Nightly Release
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *' ## Every day at midnight UTC
+
+permissions:
+  contents: read
+
+jobs:
+  checkout:
```

Comment:
is this checkout needed as a separate job instead of just including as the first step in the test job? noticed this in `release.yaml` as well.

Reply:
I used `parallel-tests.yaml` as the example here. I believe it is necessary because in order to call the test job, you have to need to be able to reference it from a local repo. Then the test job needs the checkout because its in a new container <- my understanding of GH Actions works

Reply:
good call, updated 🚀